sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/socket.scala

```diff
@@ -136,7 +149,8 @@ class TextSocketSourceProvider extends StreamSourceProvider with DataSourceRegis
       parameters: Map[String, String]): Source = {
     val host = parameters("host")
     val port = parameters("port").toInt
-    new TextSocketSource(host, port, sqlContext)
+    new TextSocketSource(host, port,
+      parameters.getOrElse("includeTimestamp", "false").toBoolean, sqlContext)
```

Comment:
but its better to have single function that catches that error and prints a better error message like "value of option includeTimestamp cannot be parsed. allowed values are "true" or "false".
also it needs to be a IllegalArgumentException